File: HackerRank/Practice/Algorithms/Warmup/DiagonalDifference/java/DiagonalDifference.java
/*
 * Solution to the "Diagonal Difference" challenge from HackerRank:
 * https://www.hackerrank.com/challenges/diagonal-difference/problem
 * Created on: 21.03.2020
 * Last modified: 21.03.2020
 * Author: Adam Graliński (adam@gralin.ski)
 * License: MIT
 */

import java.io.*;
import java.util.*;

public class DiagonalDifference
{
  private static final Scanner scanner = new Scanner(System.in);

  public static void main(String[] args) {
    int N = scanner.nextInt();
    int[][] array = new int[N][N];
    for (int row = 0; row < N; ++row) {
      for (int col = 0; col < N; ++col) {
        array[row][col] = scanner.nextInt();
      }
    }

    int diag1 = 0, diag2 = 0;
    for (int k = 0; k < N; ++k) {
      diag1 += array[k][k];
      diag2 += array[k][N - k - 1];
    }

    System.out.println(Math.abs(diag1 - diag2));
  }
}

File: HackerRank/Practice/Algorithms/Warmup/DiagonalDifference/cpp/DiagonalDifference.cpp
/**
 * Solution to the "Diagonal Difference" challenge from HackerRank:
 * https://www.hackerrank.com/challenges/diagonal-difference/problem
 * Created on: 21.03.2020
 * Last modified: 21.03.2020
 * Author: Adam Graliński (adam@gralin.ski)
 * License: MIT
 */

#include <iostream>
#include <vector>

int main(int, char**)
{
  std::size_t N;
  std::cin >> N;

  std::vector<std::vector<int>> v;
  v.resize(N);
  for (std::size_t row = 0; row < N; ++row) {
    v[row].resize(N);
    for (std::size_t col = 0; col < N; ++col) {
      std::cin >> v[row][col];
    }
  }

  int diag1{}, diag2{};
  for (std::size_t k = 0; k < N; ++k) {
    diag1 += v[k][k];
    diag2 += v[k][N-k-1];
  }

  std::cout << std::abs(diag1 - diag2) << "\n";
  return 0;
}
